genTriposGAFFandGAFF2.py

```diff
@@ -27,48 +26,55 @@ def GenTriposGAFF(mol):
-    openmoltools.amber.run_tleap(molName, 'gaff2_mol2/%s.mol2' % molName, 'gaff2_mol2/%s.frcmod' % molName, prmtop_filename = 'gaff2_mol2/%s.prmtop' % molName, inpcrd_filename = 'gaff2_mol2/%s.inpcrd' % molName, leaprc = 'leaprc.gaff2')
-
-    # generate gromacs .top and .gro files
-    #openmoltools.utils.convert_via_acpype( molName, prmtop_filename, inpcrd_filename)
-
+    if not (os.path.exists(inpcrd_filename) and os.path.exists(prmtop_filename)):
+        # calculate partial charges
+        try:
+            chgmol = openmoltools.openeye.get_charges(mol, keep_confs=1)
+        except RuntimeError, e: # catch specified warning or Runtime error
+            print("Skipping %s in %s due to RuntimeError." % (mol.GetTitle(), inputfile))
+            return False
+        except ValueError, e:
+            print("Skipping %s in %s due to ValueError." % (mol.GetTitle(), inputfile))
+            return False
+        # reset the title to original title (get_charges renames mol)
+        chgmol.SetTitle(mol.GetTitle())
+
+        # generate tripos mol2 files
+        openmoltools.openeye.molecule_to_mol2(chgmol, tripos_filename)
+
+        # generate the gaff mol2 file and the .frcmod file
+        _, _ = openmoltools.amber.run_antechamber(molName, tripos_filename,charge_method=None,\
+         gaff_mol2_filename=gaff_mol2_filename, frcmod_filename=frcmod_filename)
+        # generate the gaff .inpcrd and .prmtop files
+        openmoltools.amber.run_tleap(molName, gaff_mol2_filename, frcmod_filename,\
+         prmtop_filename=prmtop_filename, inpcrd_filename=inpcrd_filename)
+    else:
+        print("Already found gaff and tripos files for %s. Skipping." % molName)
+    gaff2_inpcrd=('./gaff2_mol2/%s.inpcrd' % molName)
+    gaff2_prmtop=('./gaff2_mol2/%s.prmtop' % molName)
+    #generate gaff2 mol2 and frcmod files
+    if not (os.path.isfile(gaff2_inpcrd) and os.path.isfile(gaff2_prmtop)):
+        os.system('antechamber -i %s/%s.mol2 -fi mol2 -o %s/%s.mol2 -fo mol2 -at gaff2'\
+         % ('tripos_mol2', molName, 'gaff2_mol2', molName))
+        os.system('parmchk2 -i %s/%s.mol2 -f mol2 -s gaff2 -o %s/%s.frcmod'\
+         % ('gaff_mol2', molName, 'gaff2_mol2', molName))
+
+        # generates gaff2 inpcrd and prmtop files
+        try:
+            openmoltools.amber.run_tleap(molName, 'gaff2_mol2/%s.mol2' % molName,\
+             'gaff2_mol2/%s.frcmod' % molName, prmtop_filename = 'gaff2_mol2/%s.prmtop'\
+             % molName, inpcrd_filename = 'gaff2_mol2/%s.inpcrd' % molName, leaprc = 'leaprc.gaff2')
+        except IOError, e:
+            print ("Skipping %s in %s due to IOError." % (mol.GetTitle(), inputfile))
+            return False
+        except RuntimeError, e:
+            print ("Skipping %s in %s due to RuntimeError for GAFF2." % (mol.GetTitle(), inputfile))
+            return False
+    else:
+        print("Already found GAFF2 files for %s. Skipping." % molName)
     return True
-    
+
```

Comment:
This is a lot of changes, I'll let @vtlim comment on this section as she's more familiar with what you're changing.